lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/IFileDeSerializer.cs

```diff
@@ -30,6 +34,17 @@ public interface IFileDeSerializer<T>
         /// </summary>
         /// <param name="fileFolder"></param>
         /// <returns></returns>
+        [Obsolete("Remove after 0.14")]
         T Deserialize(string fileFolder);
+
+        /// <summary>
+        /// The input is a set of file paths. 
+        /// The output is of type T which is defined by the client.
+        /// If there is any IO error, IOException could be thrown.
+        /// </summary>
+        /// <param name="filePaths"></param>
+        /// <param name="local"></param>
+        /// <returns></returns>
+        T Deserialize(ISet<string> filePaths, bool local);
```

Comment:
Shouldn't this boolean be passed in as a parameter to the constructor? Is there ever a need to use one and the same instance of `IFileDeSerializer` where it sometimes downloads first and sometimes doesn't?